### DISCUSSION TITLE
Keycloak >> Elasticsearch >> APM Services (Kibana)

### DISCUSSION THREAD

**Discussion by rkazma** [keycloak/keycloak#28474] [Q&A]
i have installation folder keycloak 22.0.3.
i have identity-server repo which uses keycloak solution.
i use the repo by running maven clean install, then copy the jar file from path\to\identity-server\{projectname}\target and paste it in path\to\keycloak-22.0.3\providers.
then i open command prompt and use the command cd path\to\keycloak-22.0.3\bin
then i run the command : kc.bat start-dev
note that i already installed elasticsearch, apm server, kibana and opentelemetrycollector containers and i can run it using localhost:5601, and i have multiple .net solutions that are configured to display the traces. 
i want keycloak traces under APM >> services

what should be done here to start seeing the service instance under the APM >> Services and start seeing the transactions, logs etc.?

**Comment by codespearhead** [comment]
It seems that your question has more to do with the [monitoring of Java applications with Elastic-vendored solutions](https://www.elastic.co/guide/en/apm/agent/java/current/opentelemetry-bridge.html) than with Keycloak itself, in which case I suggest you close this discussion and ask it there. By the way, [this Keycloak extension](https://docs.elastic.co/en/integrations/keycloak) [is not an official Elastic integration](https://docs.elastic.co/integrations/support#community-developed).

[Elastic Observability](https://www.elastic.co/guide/en/observability/current/observability-introduction.html), just like Keycloak, [natively supports OpenTelemetry](https://www.elastic.co/guide/en/observability/current/apm-open-telemetry-direct.html), which is a set of APIs, libraries, agents, and instrumentation that provide an open-source and vendor-neutral way to collect and export telemetry data.

So you must [enable metrics in Keycloak](https://www.keycloak.org/server/configuration-metrics) and do it the OpenTelemetry way as exemplified [in this repository](https://github.com/jangaraj/keycloak-with-opentelemetry).
